XMonad/Prompt.hs

```diff
@@ -631,6 +632,26 @@ eventLoop handle stopAction = do
     handle (keysym,keystr) event
     stopAction >>= flip unless (eventLoop handle stopAction)
 
+{- Note [Allow Events]
+
+Some settings (like @clickJustFocuses = False@) set up the passive
+pointer grabs that xmonad makes to intercept clicks to unfocused windows
+with @pointer_mode = grabModeSync@ and @keyboard_mode = grabModeSync@.
+This means that any click in an unfocused window leads to a
+pointer/keyboard grab that freezes both devices until 'allowEvents' is
+called. But "XMonad.Prompt" has its own X event loop, so 'allowEvents'
+is never called and everything remains frozen indefinitely.
+
+This does not happen when the grabs are made with @grabModeAsync@, as
+pointer events processing is not frozen and the grab only lasts as long
+as the mouse button is pressed.
+
+Hence, in this situation we call 'allowEvents' in the prompts event loop
+whenever a button event is received, releasing the pointer grab. In this
```

Comment:
We should either only call it for the button event (in `handleOther`, probably) or mention that we always call it because it's safe (and it probably is: [“The request has no effect if the pointer is not grabbed by the client or if the pointer is not frozen as the result of an event.”](https://tronche.com/gui/x/xlib/input/XAllowEvents.html))